ivadomed/training.py

```diff
@@ -54,6 +55,22 @@ def train(model_params, dataset_train, dataset_val, training_params, path_output
     # Write the metrics, images, etc to TensorBoard format
     writer = SummaryWriter(log_dir=path_output)
 
+    # Only initialize wandb if required params are found in the config file
+    if wandb_params is not None and wandb_params[WandbKW.WANDB_API_KEY]:
+        # Collect all hyperparameters into a dictionary
+        cfg = { **training_params, **model_params}
+
+        # Get the actual project, group, and run names if they exist, else choose the temporary names as default
+        project_name = wandb_params.get(WandbKW.PROJECT_NAME, "temp_project")
+        group_name = wandb_params.get(WandbKW.GROUP_NAME, "temp_group")
+        run_name = wandb_params.get(WandbKW.RUN_NAME, "temp_run")
+
+        # Log on to WandB (since the API_KEY is already provided)
+        wandb.login(key=wandb_params[WandbKW.WANDB_API_KEY])
```

Comment:
I haven't tied it yet, but curious how this would work if the `config.JSON` has this key set to "" like what you have in the PR. Would it default back to system wandb login? or would it throew exception, requiring what @kanishk16 has recommended try/catch loop?

Reply:
Okay, something interesting happened here - I set-up the try/catch loop as suggested by @kanishk16 and left the wandb_api_key value in the config file empty. Even then, the code ran without breaking i.e. wandb logging was enabled. This is because WandB does not always need the API Key to work. When the user is running it for the first time ever, only the API Key is required. This is because wandb stores the key in the `~/.netrc` file locally. 

This is something useful to potentially mention in the documentation that the it would suffice if the user only uses the API Key for the first time. Then, it can be left empty (but, the "wandb_api_key" key in the dictionary cannot be deleted)